models/account_invoice.py

```diff
@@ -0,0 +1,406 @@
+# -*- coding: utf-8 -*-
+
+import logging
+import time
+from collections import namedtuple
+from datetime import datetime
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+from odoo.tools import DEFAULT_SERVER_DATETIME_FORMAT as DATETIME_FORMAT
+
+_logger = logging.getLogger(__name__)
+
+UNKNOWN_ERROR = _("Unknown error when synchronize invoice data")
+
+ERROR_MESSAGE = {
+    'duplicated_partner_with_contact_id': _(
+        "You cannot have two partners with the same civicrm Id"),
+    'invalid_parameter_type': _(
+        "Wrong CiviCRM request - invalid \"{}\" parameter "
+        "data type: {} expected {}"),
+    'missed_required_parameter': _(
+        "Wrong CiviCRM request - missed required field: {}"),
+    'lookup_id_error': _(
+        "This {} doesn't exist in ODOO: {}"),
+}
+
+LOOK_UP_MAP = {
+    'contact_civicrm_id': ('res.partner', 'x_civicrm_id', 'partner_id'),
+    'account_code': ('account.account', 'code', 'account_id'),
+    'currency_code': ('res.currency', 'name', 'currency_id'),
+    'product_code': ('product.product', 'default_code', 'product_id'),
+    'tax_name': ('account.tax', 'name', 'invoice_line_tax_ids'),
+    'journal_code': ('account.journal', 'code', 'journal_id'),
+    'invoice_civicrm_id': ('account.invoice', 'x_civicrm_id', 'x_civicrm_id'),
+    'invoice_line_civicrm_id': ('account.invoice.line', 'x_civicrm_id',
+                                'x_civicrm_id'),
+    # 'payment_civicrm_id': ('account.payment', 'x_civicrm_id', 'x_civicrm_id'),
+}
+
+
+class AccountInvoiceLine(models.Model):
+    _inherit = "account.invoice.line"
+
+    x_civicrm_id = fields.Integer(string='Civicrm Id', required=False,
+                                  help='Civicrm Id')
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.invoice"
+
+    x_civicrm_id = fields.Integer(string='Civicrm Id', required=False,
+                                  help='Civicrm Id')
+
+    @api.model
+    def civicrm_sync(self, input_params):
+        """Synchronizes CiviCRM Contributions to Odoo invoice.
+         Creates new invoice if not exists and updates is it is
+         present in Odoo. Returns back to CiviCRM assigned invoice_id and
+         update_date and data processing status.
+        """
+        _logger.debug('Start Invoice civicrm_sync')
```

Comment:
I don't think this is a clear debugging info, maybe **Start CiviCRM contribution to invoice syncing** is better message ?